specs/monitoring_integration.adoc

```diff
@@ -0,0 +1,321 @@
+to user via tendrl (https://github.com/Tendrl/specifications/pull/198).
+
+monitoring integration passes dashboard related metrics in dashboard json. Alert
+conditions for the particular dashboard are also part of dashboard json only.
+
+```
+{
+  "aliasColors": {},
+  "bars": false,
+  "dashLength": 10,
+  "dashes": false,
+  "datasource": null,
+  "fill": 1,
+  "id": 1,
+  "legend": {
+    "avg": false,
+    "current": false,
+    "max": false,
+    "min": false,
+    "show": true,
+    "total": false,
+    "values": false
+  },
+  "lines": true,
+  "linewidth": 1,
+  "links": [],
+  "nullPointMode": "null",
+  "percentage": false,
+  "pointradius": 5,
+  "points": false,
+  "renderer": "flot",
+  "seriesOverrides": [],
+  "spaceLength": 10,
+  "span": 12,
+  "stack": false,
+  "steppedLine": false,
+  "targets": [
+    {
+      "refId": "A"
+    }
+  ],
+  "thresholds": [
+    {
+      "value": 56,
+      "op": "gt",
+      "fill": true,
+      "line": true,
+      "colorMode": "critical"
+    }
+  ],
+  "timeFrom": null,
+  "timeShift": null,
+  "title": "Panel Title",
+  "tooltip": {
+    "shared": true,
+    "sort": 0,
+    "value_type": "individual"
+  },
+  "type": "graph",
+  "xaxis": {
+    "buckets": null,
+    "mode": "time",
+    "name": null,
+    "show": true,
+    "values": []
+  },
+  "yaxes": [
+    {
+      "format": "short",
+      "label": null,
+      "logBase": 1,
+      "max": null,
+      "min": null,
+      "show": true
+    },
+    {
+      "format": "short",
+      "label": null,
+      "logBase": 1,
+      "max": null,
+      "min": null,
+      "show": true
+    }
+  ],
+  "alert": {
+    "conditions": [
+      {
+        "type": "query",
+        "query": {
+          "params": [
+            "A",
+            "5m",
+            "now"
+          ]
+        },
+        "reducer": {
+          "type": "avg",
+          "params": []
+        },
+        "evaluator": {
+          "type": "gt",
+          "params": [
+            56
+          ]
+        },
+        "operator": {
+          "type": "and"
+        }
+      }
+    ],
+    "noDataState": "no_data",
+    "executionErrorState": "alerting",
+    "frequency": "60s",
+    "handler": 1,
+    "notifications": [],
+    "name": "Panel Title alert"
+  }
+}
+```
+
+
+== Alternatives
+
+None
+
+
+== Data model impact:
+
+None
+
+
+== Impacted Modules:
+
+=== Tendrl API impact:
+
+None
+
+=== Notifications/Monitoring impact:
+
+None
+
+=== Tendrl/common impact:
+
+None
+
+=== Tendrl/node_agent impact:
+
+Cluster creation or import cluster flow should install the monitoring-integration
+packages and start that as service.
+
+=== Sds integration impact:
+
+None
+
+=== Tendrl/monitoring-integration impact:
+
+Introduce a new component monitoring integration in tendrl.
+
+Monitoring integration should have all predefined json files. When monitoring
+integration starts it should start the script to configure the grafana environment
```

Comment:
@nthomas-redhat for first milestone we are creating monitoring-integration like this, if monitoring integration start then it will start configure grafana. we are not decided complete flow of monitoring integration yet